application/common/models/User.php

```diff
@@ -1003,8 +1025,46 @@ public function isPromptForMfa(): bool
         return false;
     }
 
-    public function updateExternalGroups($appPrefix, $appExternalGroups): bool
+    public static function syncExternalGroups(string $appPrefix, array $desiredExternalGroupsByUserEmail): array
```

Comment:
This could benefit from a function doc header since "sync external groups" sounds like it's going to sync data to groups that are external, i.e. outside of idp-id-broker. In contrast, it's actually pulling **in** data and updating user data. Here's a start:
```php
/*
 * syncExternalGroups updates users' `externalGroups` attribute with data in the
 * `desiredExternalGroupsByUserEmail` parameter. That parameter is indexed by user 
 * email address and contains a list of external groups to add to the `externalGroups`
 * attribute.
 */
```
It may also be helpful to rename the function to something like `updateUsersExternalGroups`.

Reply:
Ahh, good point. I started adding docs in a subsequent branch (not yet PR'd), but I appreciate the insights on helpful wording/naming. 👍